eclipse-collections-api/src/main/java/org/eclipse/collections/api/map/MutableMapIterable.java

```diff
@@ -433,4 +434,11 @@ default <K1, V1, V2> MutableMapIterable<K1, V2> aggregateBy(
                 valueFunction.valueOf(value)));
         return map;
     }
+
+    @Override
+    default void forEach(BiConsumer<? super K, ? super V> action)
+    {
+        Objects.requireNonNull(action);
+        this.forEachKeyValue(action::accept);
+    }
```

Comment:
In my opinion, this implementation should be pulled up to MapIterable. So the version here would be:

```java
    @Override
    default void forEach(BiConsumer<? super K, ? super V> action)
    {
        MapIterable.super.forEach(action);
    }
```

Reply:
This could be done in a separate PR. `MapIterable` does not extend `Map`. By putting this as a default method on `MapIterable`, it might create a challenge similar to what we faced with [`isEmpty` default methods on `CharSequence`](https://stuartmarks.wordpress.com/2020/09/22/incompatibilities-with-jdk-15-charsequence-isempty/). We understand this particular problem now, so we can fix anything that arises, but I think it would be better to do this in small steps. This PR is meant to fix the performance issue currently faced with `Map` `forEach` delegating to `entrySet` in the default implementation.

Reply:
Won't this create an entrySet()?

`Maps.immutable.empty().castToMap().forEach((k, v) -> {})`

Reply:
I stepped through in a debugger, it creates an empty UnifiedSet for the entrySet.

Reply:
Yes it will. Good point. Ok, I agree then it should be moved up to `MapIterable`.

Reply:
Thanks for the feedback @motlin and @donraab!

I was thinking about moving it up to `MapIterable` but had the same concerns as @donraab about it not being a subclass of the Map class. I agree that it does make more sense that `forEach` would have the same issues for `ImmutableMap`. I will work on moving it up.